src/Makefile.test.include

```diff
@@ -416,6 +416,31 @@ if EMBEDDED_UNIVALUE
 	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -C univalue check
 endif
 
+if ENABLE_BENCH
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY)
+endif
+else
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY)
+endif
+endif
+	@echo "Using the valgrind memory error detector: expect an ~50x slowdown, valgrind 3.14 or later required"
+	@echo "Running $(TEST_BINARY) under valgrind -- this will take a quite a while ..."
+	valgrind --suppressions=$(top_builddir)/contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 --quiet $(TEST_BINARY)
+if ENABLE_BENCH
+	@echo "Running $(BENCH_BINARY) -evals=1 -scaling=0 under valgrind -- this will take quite a while ..."
```

Comment:
nits noticed while testing on the command line:

- I think "quite a" is a bit verbose here and in line 440 -- "this may take a while" on its own is fine

- s/while .../while.../

Reply:
Yes, that's better. Thanks! Now updated. Please re-review :)